File: week03/week03-5.py
#week03-5.py
#Leetcode 459
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
      s2 = s + s
      s2 = s2[1:len(s2)-1]
      # print(s2)
      if s in s2: return True
      else: return False  

File: week03/week03-5.cpp
// week03-5.cpp
// Leetcode 459
// abcabcabcabc
// abcabcabcabcabcabcabcabc 把他copy兩次
// abcabcabcabcabcabcabcabc 頭尾砍掉1個字母
class Solution {
public:
    bool repeatedSubstringPattern(string s) {
        string s2 = s + s;
        int N = s2.length();
        return s2.substr(1,N-2).find(s) != string::npos;
    }
};